Boost telemetry to 1:2 when transferring MSP

CapnBry opened the pull request:
This PR boosts the TLM ratio to 1:2 following an MSP packet send, to dedicate more bandwidth to the downlink for the response. Credit for the idea of only switching when the BF.lua is active goes to @schugabe I believe. This code is always active (i.e. no user_define for it) since it only triggers if there is MSP data, which means the user wants it.

Ideally, we'd wait until the MSP transfer is complete **then** boost the telemetry (since a higher telemetry rate actually reduces our uplink MSP bandwidth which is detrimental to the request), but this way is simple and has the side effect of allowing MSP commands to work even when the TLM Ratio configured is Off. Telemetry actually turns on, the request completes, and telemetry turns back off. If we find ourselves needing more upstream bandwidth as well, we can make it more complicated at that time.

Sample of 250Hz arming and disarming, not a single packet missed:
![image](https://user-images.githubusercontent.com/677183/129457101-00a5b482-8df9-4b93-93b5-c12a06f07ccf.png)

## Packet sequence
* MSP data packet
* If ratio is not already 1:2: Send sync packet with TLM Ratio set to 1:2
* Sequence will be:
  * Telemetry
  * Channels
  * Telemetry
  * MSP (if any left to send, else Channels again)
* The TLM ratio will remain 1:2 until the next scheduled sync packet. This is 4 seconds currently which is a ton of time to complete the transfer. It will always use the "connected" sync timing regardless of if the TLM ratio was on or off, because the TLM is turned on by the process.

### Bug fixed
* Lua config was using the current power, air rate, and TLM ratio as the configured value and should have been using the configured value for all of them, so I fixed that. It was preventing me from being able to test changing TLM ratios when not armed.

# Alternative Option (previous commit)
This PR used to switch the TLM Ratio to 1:2 when AUX1 is low (disarmed) and `-DTLM_DISARMED_BOOST` is enabled. Requested in #654

It is a simple check of the AUX1 channel when the sync packet is generated to determine what ratio to tell the RX to run at. After sending a CHANNELS packet that has a different AUX1 value, a sync spam is sent to adjust the TLM rate immediately instead of waiting a full sync duration. I've tested on BF and didn't see any change in LQ at the time of switch so it seems the RX is in good shape and transitions cleanly. I've also made sure the unboosted way (without the define) still works properly when changing ratios.

If folks think that is a better idea instead of this, I can make those changes. Both are good in my mind, but whichever is more popular is fine.

JyeSmith commented:
Something I raised in the issue is racers may have problems arming one the blocks.  Users will want lua/MSP to setup vtx etc, but once in the blocks everyone's LQ will tank.

So boost on MSP is my preference.

CapnBry commented:
> So boost on MSP is my preference.

I agree. It was only after I posted the PR and was in a conversation with Johannes that I realize it should be only on MSP transfer. I think this makes a lot more sense because that's really the only time bandwidth is needed, apart from discovering sensors in OpenTX. With the MSP way there's not even any need to have a define for it, since it only kicks in if the user is actually using MSP.

I've updated the PR with this code and removed the "on ARM" code.

pkendall64 commented on src/src/tx_main.cpp:
```diff
@@ -513,14 +528,9 @@ void sendELRSstatus()
-  #endif
+  setLuaTextSelectionValue(&luaAirRate, (uint8_t)config.GetRate());
+  setLuaTextSelectionValue(&luaTlmRate, (uint8_t)config.GetTlm());
+  setLuaTextSelectionValue(&luaPower, (uint8_t)config.GetPower());
```
Why this last change? Do we not want the currentPower level if using dynamic power?
Actually I think this is better, but we should perhaps add an info field with the current power level as part of the LUA changes perhaps. Or undo this section from this commit and add it to the LUA change patch?

CapnBry replied:
No, the configuration system is for changing the configuration, not getting the current status. The current status is available from the Telemetry system. If the user is changing a setting, then they need to see what it is currently set at, not the current status which misleads them to thinking they're *configured* to that value. If it displays the current value, then how does the user know what it is set to?

If you make this change in the Lua PR that's fine too, I'll fix up the conflict but it needs to be done somewhere because this is the only way the user can see their configured setting.

schugabe commented:
I did think about this change a bit more: Did you test how often this code switches between 1:2 and the configured ratio during the vtx tables download or some more data heavy save sequences? I can think of one problem: suppose we have a MSP transfer from tx=>rx that is finished right before the next sync package: then the tlm ratio would not be increased for this message. 

Another idea would be to switch to 1:2 when msp transfer is detected and continue in this mode until the quad is armed.

CapnBry commented:
After the telemetry switch, there's a full 4 seconds of 1:2 telemetry before it switches back (or 5 seconds if my other PR to extend it is merged). The sync timer is reset when we switch to 1:2 so there's always a guaranteed full interval and we don't have to worry about if it was just about to sync. Also, because telemetry will turn on and set the TX status to "connected" we never have to worry about the disconnectedSyncInterval shortening the transfer time. You'll always get at least 4 seconds of transfer time per boost.

I don't see any need to make it any more complicated since the Lua will re-request if a transfer times out which will re-boost the rate again if it has gone back down.

JyeSmith commented:
@CapnBry I went to merge this but there are now conflicts due to and earlier PR merge.
